### ISSUE TITLE
`must-return-ref` false positives: decl-literal constructors and nested functions

### ISSUE THREAD

**Issue by DonIsaac** [DonIsaac/zlint#373]
# `must-return-ref` false positives: decl-literal constructors and nested functions

- **Severity:** Medium (rule is on by default at `.warning`)
- **Status:** Both verified by reproduction
- **Location:** `src/linter/rules/must_return_ref.zig:97-128`

## Bug 1: returning a fresh value like `.empty` / `Type.init` is flagged

The return visitor reports **every** returned `field_access`:

```zig
// must_return_ref.zig:122
pub fn visit_return(self: *ReturnVisitor, node: Node.Index) VisitError!walk.WalkState {
    const returned = self.ast.nodeData(node).opt_node.unwrap() orelse return .Continue;
    // todo: check that leftmost ident is `this`.
    if (self.ast.nodeTag(returned) != .field_access) return .Continue;
    ctx.report(mustReturnRefDiagnostic(ctx, self.typename, returned));
```

The rule's doc-comment says new instances are fine, but the canonical way to construct an empty container since Zig 0.14 is a decl literal — which is a `field_access`:

```zig
pub fn newList() std.ArrayListUnmanaged(u32) {
    return std.ArrayListUnmanaged(u32).empty;
    // warning: must-return-ref: Members of type `ArrayListUnmanaged` must be passed by reference
}
```

Verified: flagged despite being a fresh, capacity-less value that is perfectly safe to copy. The `todo` on line 124 acknowledges the missing check.

## Bug 2: `return` statements inside nested container methods are attributed to the outer function

The walk starts at the outer function's body (`initAtNode`) but never skips nested function declarations, so a `return` in a locally-declared helper struct's method is checked against the *outer* function's return type:

```zig
pub fn makeList(self: *Foo) std.ArrayListUnmanaged(u32) {
    const Helper = struct {
        name: []const u8,
        fn getName(h: *const @This()) []const u8 {
            return h.name; // flagged against ArrayListUnmanaged
        }
    };
    ...
}
```

Verified: the warning points at `return h.name;`, a `[]const u8`. Local helper structs with methods (sort comparators, hash-map contexts) are common Zig.

## User impact

A default-on rule warning on the standard `.empty`/`.init` construction pattern and on unrelated nested-function returns erodes trust in the linter and pushes users to disable the rule.

## Proposed fix

- Bug 1: implement the todo — only report when the accessed object chains back to a parameter or local of the flagged type (at minimum, skip when the leftmost identifier of the field access resolves to a *type*, since `Type.decl_literal` is by definition a fresh value, not a member access on an instance).
- Bug 2: add `visit_fn_decl` (and `visit_test_decl`) to `ReturnVisitor` returning `.Skip` so the walker does not descend into nested function bodies.

Add pass cases for both patterns to the rule tests.
